SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosOfferScheduler.java

```diff
@@ -240,6 +238,59 @@ public SingularityMesosOfferScheduler(MesosConfiguration mesosConfiguration,
     return offerHolders.values();
   }
 
+  private Void buildScoringFuture(
```

Comment:
nit on naming, this method isn't actually building the future, it's a synchronous method for doing the actual scoring. I'd either rename this to calculateScore (or something like that), or move the supplyAsync inside this method and have it return the actual `CompletableFuture<Void>`